pkgs/top-level/all-packages.nix

```diff
@@ -41948,4 +41948,6 @@ with pkgs;
   insulator2 = callPackage ../applications/misc/insulator2 {};
 
   animdl = python3Packages.callPackage ../applications/video/animdl { };
+
+  one-click-backup = qt6Packages.callPackage ../applications/backup/nis-one-click-backup { };
```

Comment:
New packages should use the [`/pkgs/by-name`](https://github.com/NixOS/nixpkgs/blob/2565ab7b53ae6966bd08e68203a89645961849b3/pkgs/by-name/README.md) system instead of the old `all-packages.nix` one, unless they meet specific criteria that make them unsuitable for the former. Please move the package, and drop the `all-packages.nix` entry.

Reply:
Am I understanding this right? I move the package definition into another folder and delete everything I added from `all-packages.nix`? But where do I tell nix to use qt6Packages.callPackage then? This doesn't seem clear to me when reading the [documentation](https://github.com/NixOS/nixpkgs/blob/2565ab7b53ae6966bd08e68203a89645961849b3/pkgs/by-name/README.md) you gave me.

Reply:
> Am I understanding this right?

Yep.

> But where do I tell nix to use qt6Packages.callPackage then?

`libsForQt5.callPackage` / `qt6Packages.callPackage` let you import `qtbase`, `wrapQtAppsHook` etc at the top of your derivation. Whether those are from Qt5 or Qt6 will then depend on which `callPackage` you used, letting you write your derivation somewhat Qt version-agnostic.

Since you're directly using `qt6` already, `qt6Packages.callPackage` shouldn't matter.

Reply:
I'm not really understanding what you mean here. So I need to just move my derivation to another directory and then it "just works" (no need to write the `qt6packages.callPackage `anywhere)? Because every time I've used `callPackage` without the `qt6packages`, the derivation didn't build (got an error with `wrapQtAppsHook`)

Reply:
Use `qt6.wrapQtAppsHook`.

Reply:
Okay, but where do I write this line? I thought with the by-name structure you don't define such things anywhere?

Reply:
Okay, I think I got it now (finally). Commit [924be71](https://github.com/NixOS/nixpkgs/pull/287497/commits/924be717ca4ebc954288443bb4493e601c41bec3) should now use the `by-name` structure. An my test machine, it builds fine.